### ISSUE TITLE
Glooctl check connection refused

### ISSUE THREAD

**Issue by sheidkamp** [solo-io/gloo#10020]
### Gloo Edge Product

Open Source

### Gloo Edge Version

1.17+

### Kubernetes Version

any

### Describe the bug

Customer seeing:
```
~$ glooctl get proxy
Error: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 127.0.0.1:51003: connect: connection refused"
```

### Expected Behavior

`glooctl` should retry for longer.

### Steps to reproduce the bug

* Update the number of retries [here](https://github.com/solo-io/gloo/blob/main/projects/gloo/cli/pkg/common/proxy.go#L172) to 1
* run `make build-cli`
* Use the built glooctl to run check and get a connection refused error:
 _output/glooctl-darwin-arm64 check  
----------
glooctl binary version (1.0.0-sah1) differs from server components (v1.18.0-beta20) by at least a minor version.
Consider running:
_output/glooctl-darwin-arm64 upgrade --release=v1.18.0-beta20
----------

Checking Deployments... OK
Checking Pods... OK
Checking Upstreams... OK
Checking UpstreamGroups... OK
Checking AuthConfigs... OK
Checking RateLimitConfigs... OK
Checking VirtualHostOptions... OK
Checking RouteOptions... OK
Checking Secrets... OK
Checking VirtualServices... OK
Checking Gateways... OK
Checking Proxies... 7 Errors!
Checking rate limit server... OK

Skipping Kubernetes Gateway resources check -- Kubernetes Gateway integration not enabled

Skipping Gloo Instance check -- Gloo Federation not detected.
Error: 7 errors occurred:
	* rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp [::1]:60957: connect: connection refused"
	* rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp [::1]:60960: connect: connection refused"
	* rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp [::1]:60963: connect: connection refused"
	* rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp [::1]:60966: connect: connection refused"
	* rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp [::1]:60969: connect: connection refused"
	* rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp [::1]:60972: connect: connection refused"
	* rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp [::1]:60975: connect: connection refused"
```


### Additional Environment Detail

_No response_

### Additional Context

_No response_

**Comment by sheidkamp** [comment]
merged into main and 1.17.x

**Comment by sheidkamp** [comment]
Also, this is a duplicate of https://github.com/solo-io/solo-projects/issues/6785
